### ISSUE TITLE
CompositeSpanExporter should not call `spanExporter` if there are no spans to send

### ISSUE THREAD

**Issue by corecanarias** [micrometer-metrics/tracing#221]
I'm using a springboot 3 app and configured a `SpanExportingPredicate` to filter some traces based on business logic. Due to this logic, there are scenarios where the `CompositeSpanExporter.export()` method is being invoked. However, since the configured predicates may produce empty `changedSpanData`, there are cases where `spanExporter.export(changedSpanData)` is called with no data.

This results in errors for some OpenTelemetry providers, such as Honeycomb.io. To prevent such errors and avoid some additional empty requests, it would be beneficial to avoid these empty calls altogether.

My current workaround is to override the CompositeSpanExporter and check if there are spans before calling the exporter:

```java
    public CompletableResultCode export(Collection<SpanData> spans) {
        List<SpanData> changedSpanData = spans.stream().filter(this::shouldProcess).map(spanData -> {
            FinishedSpan finishedSpan = OtelFinishedSpan.fromOtel(spanData);
            for (SpanFilter spanFilter : spanFilters) {
                finishedSpan = spanFilter.map(finishedSpan);
            }
            return OtelFinishedSpan.toOtel(finishedSpan);
        }).collect(Collectors.toList());

        // Skip everything if no spans are left
        if (changedSpanData.isEmpty()) {
            return CompletableResultCode.ofSuccess();
        }

        List<CompletableResultCode> results = new ArrayList<>();
        changedSpanData.forEach(spanData -> {
            this.reporters.forEach(reporter -> {
                try {
                    reporter.report(OtelFinishedSpan.fromOtel(spanData));
                    results.add(CompletableResultCode.ofSuccess());
                }
                catch (Exception ex) {
                    results.add(CompletableResultCode.ofFailure());
                }
            });
        });
        this.exporters.forEach(spanExporter -> results.add(spanExporter.export(changedSpanData)));
        return CompletableResultCode.ofAll(results);
    }
```

**Comment by shakuzen** [comment]
Thank you for letting us know about this. Would you be interested in making a pull request to change the behavior?

**Comment by corecanarias** [comment]
Hi @shakuzen,
Yes, definitely I'll be happy to do it :D Give me some time and I'll send something 👍

**Comment by shakuzen** [comment]
Fixed by #224
